go/libraries/doltcore/env/grpc_dial_provider.go

```diff
@@ -116,7 +116,9 @@ func (p GRPCDialProvider) getRPCCreds() (credentials.PerRPCCredentials, error) {
 	if !valid {
 		return nil, nil
 	}
-	return dCreds.RPCCreds(), nil
+
```

Comment:
I think this is a private method that's being used just about (`config.WithEnvCreds` branch...), and I think we should be passing the `endpoint` we've determined from the grpcendpoint.Config which was provided to us...